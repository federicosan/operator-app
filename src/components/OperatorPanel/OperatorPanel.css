#OperatorPanel {
  max-width: 300px;
  min-width: 200px;
  background: #fdfdfd;
  flex-grow: 1;
  z-index: 4;
  box-shadow: 1px 0 1px rgba(0, 0, 0, 0.1);
  display: flex;
  flex-direction: column;
}

.OperatorPanel__header {
  display: flex;
  flex-direction: column;
  height: 50px;
  justify-content: center;
  align-self: center;
  font-family: courier;
  letter-spacing: 2px;
  text-transform: uppercase;
  font-size: 12px;
  color: #afafaf;
}


.OperatorPanel__footer {
  font-size: 10px;
  color: #777;
  text-align: center;
  padding: 1rem;
}
