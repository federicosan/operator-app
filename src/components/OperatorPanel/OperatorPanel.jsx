import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import Panel from '../Panel/Panel.jsx';
import TitleBar from '../TitleBar/TitleBar.jsx';
import OperatorProfile from '../OperatorProfile/OperatorProfile.jsx';
import OperatorClientMenu from '../OperatorClientMenu/OperatorClientMenu.jsx';
import OperatorSettingsMenu from '../OperatorSettingsMenu/OperatorSettingsMenu.jsx';

export class OperatorPanelComponent extends Panel {
  // constructor (props) {
  //   super(props);
  //
  //   this.state = {};
  // }

  render () {
    return (
      <div className="panel">
        <div className="top container">
          <TitleBar />
          <OperatorProfile />
          <OperatorClientMenu />
        </div>
        <div className="bottom container">
          <OperatorSettingsMenu />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  state
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});

const OperatorPanel = connect(
  mapStateToProps,
  mapDispatchToProps,
)(OperatorPanelComponent);

export default OperatorPanel;
