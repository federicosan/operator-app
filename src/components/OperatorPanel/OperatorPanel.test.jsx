import React from 'react';

import { shallow } from 'enzyme';

import OperatorPanel from './OperatorPanel.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ })),
};

describe('OperatorPanel', () => {
  it('matches snapshot', () => {
    const component = shallow(<OperatorPanel store={store} />);

    expect(component).toMatchSnapshot();
  });
});
