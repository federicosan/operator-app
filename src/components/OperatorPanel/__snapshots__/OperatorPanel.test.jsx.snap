exports[`OperatorPanel matches snapshot 1`] = `
<Connect(PanelComponent)
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
